intake/catalog/entry.py

```diff
@@ -80,11 +81,12 @@ def __call__(self, persist=None, **kwargs):
             s2 = s.get_persisted()
             met = s2.metadata
             if persist == 'always' or not met['ttl']:
-                return s2
-            if met['ttl'] < time.time() - met['timestamp']:
-                return s2
+                s = s2
+            elif met['ttl'] < time.time() - met['timestamp']:
+                s = s2
             else:
-                return store.refresh(s2)
+                s = store.refresh(s2)
+        s._entry = weakref.ref(self)
```

Comment:
Why store a weakref?  If we are going to require the entry to still be around to clone a `DataSource` (maybe with different kwargs), it seems like a trap to make that working contingent on something else holding a cache of entries.

Reply:
I wanted to avoid a circular reference, but actually the entry doesn't refer to the source, so this was unnecessary.

Reply:
That makes sense.  I can see how it would be useful for the entry to hold on to weakrefs of the `DataSources`.